ztp/gitops-subscriptions/argocd/PowerConfiguration.md

```diff
@@ -0,0 +1,190 @@
+
+```yaml
+metadata:
+  annotations:
+    cpu-c-states.crio.io: "disable"
+    cpu-freq-governor.crio.io: "performance"
+```
+
+It is worthwhile to note the following:
+
+- These annotations are likely to be used with the other high performance cri-o annotation described [here](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/cnf-low-latency-tuning.html#node-tuning-operator-pod-power-saving-config_cnf-master).
+- These annotations have the same restrictions as the other high performance cri-o annotations:
+    - The pod must use the performance-&lt;profile-name&gt; runtime class.
+    - The pod must have a [QoS Class of guaranteed](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed) (i.e. uses whole CPUs).
+
+## Migrating from one mode to another
+
+This section outlines the steps and configurations for migrating from one mode to another mode.
+
+### 1) Performance to Power saving
+
+Migrating from a Performance configuration to a Power saving configuration requires setting the workloadHints parameters as follows:
+
+```yaml
+workloadHints:
+  realTime: true
+  highPowerConsumption: false
+  perPodPowerManagement: true
+```
+
+> **Note**: Critical workload pods should be annotated with the [crio annotations](#optional-power-saving-mode) shown above prior to the migration.
+
+It is recommended to set a cpu governor in the additional kernel arguments specification as well as capping the max cpu frequency as explained in the [power saving mode](#3-power-saving-mode).
+
+After the above policy changes have been made, it should be pushed to the Git repository and synced through argocd policy application. If the policy has a `remediationAction` set to `enforce`, then TALM (topology aware lifecycle manager) should do the needful of applying the policy to the effected SNOs. If however the `remediationAction` is configured for `inform`, then a CGU (cluster group upgrade) CR must be created and applied to the hub cluster.
+
+After the policy has been applied to the effected SNOs, the change can be verified by viewing the PerformanceProfile on the SNOs:
+
+```sh
+oc get PerformanceProfile -o yaml
+```
+
+### 2) Power saving to Performance
+
+Migrating from a Power saving configuration to a [Performance (low latency)](#1-performance-mode-low-latency-default) configuration requires setting the workloadHints parameters as follows:
+
+```yaml
+workloadHints:
+  realTime: true
+  highPowerConsumption: false
+  perPodPowerManagement: false
+```
+
+If a cpu governor is configured via the additional kernel arguments, this must be removed according to the `complianceType` of the policy. If the `complianceType` is configured to `mustonlyhave`, then this can be accomplished by setting the `additionalKernelArgs` to the original list as defined in the `ztp/source-crs/PerformanceProfile.yaml` file. This is shown below:
```

Comment:
```suggestion
If a cpu governor is configured via the additional kernel arguments, and the configuration is managed through Policy, this must be removed according to the `complianceType` of the policy. If the `complianceType` is configured to `mustonlyhave`, then this can be accomplished by setting the `additionalKernelArgs` to the original list as defined in the `ztp/source-crs/PerformanceProfile.yaml` file. This is shown below:
```